src/AutoUI/Filters/Filters.tsx

```diff
@@ -96,6 +106,159 @@ const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title,
+			description,
+			items: constructSchemaProperties(
+				schemaOrProperties.items as JSONSchema,
+				restRefScheme,
+				description,
+			),
+		};
+	}
+	if (firstRefSchemeKey) {
+		const property = (
+			schemaOrProperties as NonNullable<JSONSchema['properties']>
+		)[firstRefSchemeKey as keyof typeof schemaOrProperties];
+		if (!property) {
+			throw new Error(
+				`ERROR: x-ref-scheme/x-foreign-scheme bad declaration, key ${firstRefSchemeKey} not found`,
+			);
+		}
+		return {
+			[firstRefSchemeKey]: constructSchemaProperties(
+				property,
+				restRefScheme,
+				description,
+				title,
+			),
+		};
+	}
+
+	return schemaOrProperties;
+};
+
+/**
+ * Modifies a schema to apply special processing rules, such as handling ref schemes.
+ */
+const modifySchemaWithRefSchemes = (schema: JSONSchema): JSONSchema => {
+	const applyRefSchemeModifications = (
+		properties: JSONSchema['properties'] | undefined,
+	): JSONSchema['properties'] | undefined => {
+		if (!properties) {
+			return undefined;
+		}
+
+		const newProperties: JSONSchema['properties'] = {};
+		for (const [key, value] of Object.entries(properties)) {
+			if (typeof value === 'boolean') {
+				newProperties[key] = value;
+				continue;
+			}
+			const description = parseDescription(value);
+			const refScheme =
+				description?.['x-foreign-key-scheme'] || description?.['x-ref-scheme'];
+			delete description?.['x-foreign-key-scheme'];
+			delete description?.['x-ref-scheme'];
+
+			if (refScheme?.length) {
+				for (const xRefScheme of refScheme) {
+					const transformedXRefScheme =
+						convertRefSchemeToSchemaPath(xRefScheme);
+					const title = getRefSchemeTitle(transformedXRefScheme, value);
+					const entireRefScheme =
+						refSchemePrefix(value) + transformedXRefScheme;
+					if (transformedXRefScheme && description) {
+						description['x-ref-scheme'] = [transformedXRefScheme];
+					}
+					const propertyKey =
+						refScheme.length > 1
+							? `${key}${X_FOREIGN_KEY_SCHEMA_SEPARATOR}${xRefScheme}`
+							: key;
+					newProperties[propertyKey] = constructSchemaProperties(
+						value,
+						entireRefScheme?.split('.'),
+						JSON.stringify(description),
+						title,
+					);
+				}
+			} else {
+				newProperties[key] = value;
+			}
+		}
+
+		return newProperties;
+	};
+
+	return schema.properties
+		? { ...schema, properties: applyRefSchemeModifications(schema.properties) }
+		: schema;
+};
+
+/**
+ * Removes separators added to denote x-ref-scheme modifications from filter names.
+ */
+const removeRefSchemeSeparatorsFromFilters = (
+	filters: JSONSchema[],
+): JSONSchema[] => {
+	const recursivelyRemoveSeparators = (filter: JSONSchema): JSONSchema => {
+		return {
+			...filter,
+			anyOf: filter.anyOf?.map((f) => {
+				if (isJSONSchema(f) && f.anyOf) {
+					return recursivelyRemoveSeparators(f);
+				}
+				if (!isJSONSchema(f) || !f.properties) {
+					return f;
+				}
```

Comment:
nit:
```suggestion
				if (isJSONSchema(f) {
				if (f.anyOf) {
					return recursivelyRemoveSeparators(f);
				}
				} else {
				if (!f.properties) {
					return f;
				}
				}
```

Reply:
let me think about this, as I'm really against nested if, they make the readability really complex for not reason

Reply:
Forget this comment, I missed the `!` before the second if's `isJSONSchema`